src/ansys/fluent/core/launcher/error_handler.py

```diff
@@ -32,11 +33,14 @@ class UnexpectedKeywordArgument(TypeError):
     pass
 
 
-class DockerContainerLaunchNotSupported(SystemError):
-    """Raised when Docker container launch is not supported."""
+class DockerContainerLaunchError(docker.errors.DockerException):
+    """Raised when Docker container launch is not successful."""
 
     def __init__(self):
-        super().__init__("Python Docker SDK is unsupported on this system.")
+        try:
+            _ = docker.from_env()
```

Comment:
`check_docker_support()` already uses `docker.from_env()`, that is where the exception would be raised from. In my understanding we should not need to execute `docker.from_env()` again here, we need to update `check_docker_support()` instead so that exceptions are properly propagated from there

Reply:
@raph-luc 

We already tried this approach but after looking at it's usage in tests we have not changed it.

```
def check_docker_support():
    """Check whether Python Docker SDK is supported by the current system."""
    import docker

    try:
        _ = docker.from_env()
    except docker.errors.DockerException as ex:
        return False, str(ex)
    return True, ""

class DockerContainerLaunchError(SystemError):
    """Raised when Docker container launch is not supported."""

    def __init__(self, error_message):
        super().__init__(error_message)
        
docker_support, error_message = check_docker_support()

  if docker_support:
      fluent_launch_mode = LaunchMode.CONTAINER
  else:
      raise exceptions.DockerContainerLaunchError(error_message)        
```

Reply:
We can decide which approach is better. Currently implemented in the PR or mentioned in above comment.

Reply:
I am thinking that now that we have the more well-defined Fluent launcher modes with the "launcher factory" approach, we should no longer need `check_docker_support()` at all. If the user tries to launch a Fluent Docker container and for some reason it fails, appropriate exceptions should be raised and not parsed by `check_docker_support()`.

Instead of `check_docker_support()`, the start of containers can be managed through the existing `start_container` and `container_dict` arguments, and the `PYFLUENT_LAUNCH_CONTAINER` env var for testing.

I don't think there is any reason to keep `check_docker_support()` in the code right now.

Reply:
Ok. Lets see what other team members think. Thanks for the suggestion.

Reply:
Done. Thank you.